tools/rostopic/src/rostopic/__init__.py

```diff
@@ -1119,42 +1119,58 @@ def _rostopic_list_bag(bag_file, topic=None):
                     break
 
 def _sub_rostopic_list(master, pubs, subs, publishers_only, subscribers_only, verbose, indent=''):
-    def topic_type(t, topic_types):
-        matches = [t_type for t_name, t_type in topic_types if t_name == t]
-        if matches:
-            return matches[0]
-        return 'unknown type'
-
     if verbose:
         topic_types = _master_get_topic_types(master)
 
         if not subscribers_only:
             print("\n%sPublished topics:"%indent)
-            for t, l in pubs:
-                if len(l) > 1:
-                    print(indent+" * %s [%s] %s publishers"%(t, topic_type(t, topic_types), len(l)))
+            for t, ttype, tlist in pubs:
+                if len(tlist) > 1:
+                    print(indent+" * %s [%s] %s publishers"%(t, ttype, len(tlist)))
                 else:
-                    print(indent+" * %s [%s] 1 publisher"%(t, topic_type(t, topic_types)))                    
+                    print(indent+" * %s [%s] 1 publisher"%(t, y))                    
 
         if not publishers_only:
             print(indent)
             print(indent+"Subscribed topics:")
-            for t,l in subs:
-                if len(l) > 1:
-                    print(indent+" * %s [%s] %s subscribers"%(t, topic_type(t, topic_types), len(l)))
+            for t, ttype, tlist in subs:
+                if len(tlist) > 1:
+                    print(indent+" * %s [%s] %s subscribers"%(t, ttype, len(llist)))
                 else:
-                    print(indent+" * %s [%s] 1 subscriber"%(t, topic_type(t, topic_types)))
+                    print(indent+" * %s [%s] 1 subscriber"%(t, y))
         print('')
     else:
         if publishers_only:
-            topics = [t for t,_ in pubs]
+            topics = [t for t,_,_ in pubs]
         elif subscribers_only:
-            topics = [t for t,_ in subs]
+            topics = [t for t,_,_ in subs]
         else:
-            topics = list(set([t for t,_ in pubs] + [t for t,_ in subs]))                
+            topics = list(set([t for t,_,_ in pubs] + [t for t,_,_ in subs]))                
         topics.sort()
         print('\n'.join(["%s%s"%(indent, t) for t in topics]))
 
+def get_topic_list(master=rosgraph.Master('/rostopic')):
```

Comment:
I actually ran into a small issue with this just today— it's a bit of a corner case, but when `ROS_MASTER_URI` _isn't_ set, this bombs out on import, for example:

```
$ unset ROS_MASTER_URI
administrator@vm-mpurvis-xenial-ws:~$ python
Python 2.7.12 (default, Nov 19 2016, 06:48:10) 
[GCC 5.4.0 20160609] on linux2
Type "help", "copyright", "credits" or "license" for more information.
>>> import rostopic
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "workspace/lib/python2.7/dist-packages/rostopic/__init__.py", line 1152, in <module>
    def get_topic_list(master=rosgraph.Master('/rostopic')):
  File "workspace/lib/python2.7/dist-packages/rosgraph/masterapi.py", line 100, in __init__
    self._reinit(master_uri)
  File "workspace/lib/python2.7/dist-packages/rosgraph/masterapi.py", line 113, in _reinit
    raise ValueError("ROS master URI is not set")
ValueError: ROS master URI is not set
>>> 
```

I believe this will be corrected if you instead do:

```
def get_topic_list(master=None):
    if not master:
        master = rosgraph.Master('/rostopic')
```

Other than this, I'd be happy to merge this to mainline.

Reply:
Thanks for the fix. Please also address the style concern from @dirk-thomas and then this is good to go in.